examples/polynomial/fun/poly.py

```diff
@@ -0,0 +1,7 @@
+def polynomial(coefficients: dict) -> tuple:
```

Comment:
Should we name the module `function_steps` instead, just to make it clear what it actually contains?